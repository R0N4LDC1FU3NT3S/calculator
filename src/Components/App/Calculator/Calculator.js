import React, { Component } from "react"
import "./Calculator.css"
import Display from "./Display/Display"

class Calculator extends Component {
	state = {
		displayValue: "0", // value to be displayed in <Display />
		numbers: [], // values to be displayed in number <Keys />
		operators: [], // values to be displayed in operator <Keys />
		selectedOperator: "", // operator selected for math operation
		storedValue: "" // stored value to use for math operation
	}

	callOperator = () => {
		console.log("call operation")
	}

	setOperator = () => {
		console.log("set operation")
	}

	updateDisplay = () => {
		console.log("update display")
	}

	render = () => {
		// unpacked the component state by using Object Destructuring
		const { displayValue } = this.state

		return (
			<div className="calculator-container">
				<Display displayValue={displayValue} />
			</div>
		)
	}
}

export default Calculator
