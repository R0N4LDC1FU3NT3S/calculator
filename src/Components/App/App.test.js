import { shallow } from "enzyme"
import App from "./App"
import Calculator from "./Calculator/Calculator"

// ══════════════════//
//		UI test		//
// ══════════════════//

describe("App", () => {
	let wrapper

	beforeEach(() => (wrapper = shallow(<App />)))

	it("#0. Snapshot", () => {
		expect(wrapper).toMatchSnapshot()
	})

	it("#1. Render correctly", () => {
		expect(wrapper.find("div").length).toEqual(1)
	})

	it("#2. Contain Calculator component", () => {
		expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true)
	})
})
