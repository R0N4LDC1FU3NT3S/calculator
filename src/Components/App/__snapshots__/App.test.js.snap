// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App #0. Snapshot 1`] = `ShallowWrapper {}`;
